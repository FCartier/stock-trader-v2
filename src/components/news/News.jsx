import React from "react";

import { SectionHeader } from "../reusables/SectionHeader";


const News = props => {
  const { news } = props;

  return news.length ? (
    <div>
      <div className="news">
        <SectionHeader title="NEWS" />
        <ul className="news-list">
          {news.map((article, index) => {
            return (
              <li key={index}>
                <p className="article-header">
                  <a href={article.url}>{article.headline}</a>
                </p>
                <p className="article-footer">{article.source}</p>
              </li>
            );
          })}
        </ul>
      </div>
    </div>
  ) : null;
};

export default News;
