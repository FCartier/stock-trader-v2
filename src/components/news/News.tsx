import * as React from "react";

import { SectionHeader } from "../reusables/SectionHeader.tsx";
import { NewsItem, NewsList } from './types';

const News: React.SFC<NewsList> = ({news}) => {
  console.log(news);


  return renderCondition(news) ? (
    <div>
      <div className="news">
        <SectionHeader title="NEWS" />
        <ul className="news-list">
          {news
          .slice(0, 5)
          .map((article: NewsItem, index: number) => {
            return (
              <li key={index}>
                <p className="article-header">
                  <a href={article.url}>{article.headline}</a>
                </p>
                <p className="article-footer">{article.source}</p>
              </li>
            );
          })}
        </ul>
      </div>
    </div>
  ) : null;
};

function renderCondition(news: []) {
  console.log(news.length)
  if (news.length > 0) {
    return true
  }
}

export default News;
