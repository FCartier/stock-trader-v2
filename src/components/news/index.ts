import NewsContainer from "./newsContainer"

export default NewsContainer