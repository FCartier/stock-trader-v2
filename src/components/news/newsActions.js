export const FETCH_NEWS_SUCCESS = "FETCH_NEWS_SUCCESS";

