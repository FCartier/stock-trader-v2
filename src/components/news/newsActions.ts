export const FETCH_NEWS_SUCCESS = "FETCH_NEWS_SUCCESS";
export type FETCH_NEWS_SUCCESS = typeof FETCH_NEWS_SUCCESS;

export const FETCH_NEWS_FAILED = "FETCH_NEWS_FAILED";
export type FETCH_NEWS_FAILED = typeof FETCH_NEWS_FAILED;