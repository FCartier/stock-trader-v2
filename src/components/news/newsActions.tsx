import { Action, ActionCreator } from 'redux'
import { NewsList, NewsItem } from './types';
  
export const NEWS_ACTION = "NEWS ACTION";
export type NEWS_ACTION = typeof NEWS_ACTION;
  
export interface NewsAction extends Action {
  type: NEWS_ACTION
  news: NewsItem[]
}
  
export const newsAction: ActionCreator<NewsAction> = (news: NewsItem[]) => ({
  type: NEWS_ACTION,
  news
})
  