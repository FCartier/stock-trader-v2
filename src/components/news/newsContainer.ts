import News from "./News";
import { connect } from "react-redux";
import { GlobalState } from "../../StoreTypes";

const mapStateToProps = ({ news: { news, status } }: GlobalState) => ({
  news,
  status
});

export default connect(
  mapStateToProps,
  null
)(News);
