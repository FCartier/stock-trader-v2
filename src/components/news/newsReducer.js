import { FETCH_NEWS_SUCCESS } from "./newsActions"

export const news = (state = [], action) => {
    switch (action.type) {
      case FETCH_NEWS_SUCCESS:
        return action.payload;
      default:
        return state;
    }
  };