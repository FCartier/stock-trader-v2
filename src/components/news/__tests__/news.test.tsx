import * as React from "react";
import { configure, shallow } from "enzyme";
import * as Adapter from "enzyme-adapter-react-16";
import News from "../News";
import { testData } from "../__data__/news.data";
import { fetchStatus } from "../../reusables/fetchStatus";

configure({ adapter: new Adapter() });

describe("news component", () => {
  test("news component should render properly before fetch", () => {
    const component = shallow(<News news={[]} status={fetchStatus.pending} />);
    expect(component).toMatchSnapshot();
  });

  test("news component should render properly with fetch success", () => {
    const component = shallow(<News news={testData} status={fetchStatus.success} />);
    expect(component).toMatchSnapshot();
  });

  test("news component should render properly with fetch failed", () => {
    const component = shallow(<News news={testData} status={fetchStatus.failed} />);
    expect(component).toMatchSnapshot();
  });
});
