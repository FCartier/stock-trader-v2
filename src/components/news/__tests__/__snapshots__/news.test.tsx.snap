// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`news component news component should render properly before fetch 1`] = `""`;

exports[`news component news component should render properly with fetch failed 1`] = `
<div>
  <Styled(div)>
    <Component
      title="NEWS"
    />
    <NoData
      componentName="news"
    />
  </Styled(div)>
</div>
`;

exports[`news component news component should render properly with fetch success 1`] = `
<div>
  <Styled(div)>
    <Component
      title="NEWS"
    />
    <Styled(ul)>
      <li
        key="0"
      >
        <Styled(p)>
          <a
            href="https://api.iextrading.com/1.0/stock/aapl/article/5177023215303596"
          >
            Apple gets Street-high $310 price target from Wedbush
          </a>
        </Styled(p)>
        <Styled(p)>
          SeekingAlpha
        </Styled(p)>
      </li>
      <li
        key="1"
      >
        <Styled(p)>
          <a
            href="https://api.iextrading.com/1.0/stock/aapl/article/5492149748213271"
          >
            Cramer's 5 favorite tech stocks right now, including Apple, Salesforce and Adobe
          </a>
        </Styled(p)>
        <Styled(p)>
          CNBC
        </Styled(p)>
      </li>
      <li
        key="2"
      >
        <Styled(p)>
          <a
            href="https://api.iextrading.com/1.0/stock/aapl/article/7855847081717757"
          >
            2018 Third Quarter Review - Buying Dividend Growth Stocks Without Looking
          </a>
        </Styled(p)>
        <Styled(p)>
          SeekingAlpha
        </Styled(p)>
      </li>
      <li
        key="3"
      >
        <Styled(p)>
          <a
            href="https://api.iextrading.com/1.0/stock/aapl/article/7640904710004543"
          >
            Five Stocks To Buy At The Bottom
          </a>
        </Styled(p)>
        <Styled(p)>
          SeekingAlpha
        </Styled(p)>
      </li>
      <li
        key="4"
      >
        <Styled(p)>
          <a
            href="https://api.iextrading.com/1.0/stock/aapl/article/6820541619208908"
          >
            Apple: There's More In The Making
          </a>
        </Styled(p)>
        <Styled(p)>
          SeekingAlpha
        </Styled(p)>
      </li>
    </Styled(ul)>
  </Styled(div)>
</div>
`;
